src/kakarot/instructions/system_operations.cairo

```diff
@@ -581,15 +573,23 @@ namespace CallHelper {
         let stack = Stack.push(ctx.stack, status);
         let ctx = ExecutionContext.update_stack(ctx, stack);
 
-        // ret_offset, see prepare_args
-        let memory = Memory.store_n(
-            ctx.memory,
-            ctx.sub_context.return_data_len,
-            ctx.sub_context.return_data,
-            [ctx.sub_context.return_data - 1],
+        // see prepare_args
+        let ret_offset = [ctx.return_data];
+        let ret_size = ctx.return_data_len;
+        let return_data = Helpers.slice_data(
```

Comment:
does this cover the case where ret_size > ctx.sub_context.return_data_len
-> meaning does it padd with zeroes the return_data of the subcontext?

Reply:
the doc says

```
        // slice's len = min(slice_len, data_len-offset, 0)
        // which corresponds to full, partial or empty overlap with data
        // The result is zero-padded in case of partial or empty overlap.
```